### PR TITLE
#157 add support for SQ 7.2+

### PR DESCRIPTION
**Pull request by muhlba91**
Issue #157. Basic changes taken from #158, then I updated dependencies of SonarQube API. I successfully tested analysis in SonarQube 7.3, 7.4 and 7.5.
(follow-up PR from #165)

### CODE REVIEW

**Comment by romani** [comment]
https://circleci.com/gh/checkstyle/sonar-checkstyle/edit#advanced-settings
changed to:
Build forked pull requests = ON
Only build pull requests = ON
Auto-cancel redundant builds = ON


@muhlba91 ,  please change smth in commit to activate new build
for example make a comment "Issue #157: add support for SQ 6.7+"

File: checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleRulesDefinition.java [hunk 1]
```diff
@@ -60,8 +64,38 @@ static void extractRulesData(NewRepository repository, String xmlRulesFilePath,
...
                 .getResourceAsStream("/org/sonar/l10n/checkstyle.properties")) {
-            PropertyFileLoader.loadNames(repository, resource);
+            loadNames(repository, resource);
         }
         SqaleXmlLoader.load(repository, "/com/sonar/sqale/checkstyle-model.xml");
     }
+
+    private static void loadNames(NewRepository repository, InputStream stream) {
+        // code taken from https://github.com/SonarSource/sslr-squid-bridge/blob/2.5.2/
+        // src/main/java/org/sonar/squidbridge/rules/PropertyFileLoader.java#L46
```
Comment (romani) [inline checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleRulesDefinition.java]: please exlpain why sslr-squid-bridge 2.5.2 is used while in pom dependency we use 2.7.0.377

FYI: you can start line comment with `http` our checkstyle config ignore lines with URLs.

Comment (muhlba91) [reply checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleRulesDefinition.java]: This method got removed after version 2.5.2 but we need 2.7.0.377 for SQ compatibility.
Hence, I have migrated the "old" code from sslr-squid-bridge here and referenced with the last available release.

Comment (muhlba91) [reply checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleRulesDefinition.java]: Starting line comment with `http` didn't solve the line length issue. I always get line too long.

**Comment by rnveach** [comment]
CircleCI shows up now.
> ci/circleci: build — Your tests passed on CircleCI!

**Comment by muhlba91** [comment]
Great! Now all CI is green. :)
Please merge if everything looks good now.

**Comment by romani** [comment]
please make git commit comment `Issue #157: add support for SQ 6.7+`
and I will merge this PR.

Good job !

**Comment by muhlba91** [comment]
Thanks! :)

Git commit comment is changed now. Please continue with merge. When can we expect the release being available for SQ installations?

**Comment by romani** [comment]
At the end of month (26th January) we will have new checkstyle release, if all go well, at the same day we will have plugin release.
Download from SQ's plugins center/repository, available in 2 weeks after plugin release.

**Comment by Vampire** [comment]
@romani https://docs.sonarqube.org/display/PLUG/Plugin+Version+Matrix states 4.16 is compatible with <= 7.2 and no compatible version for >= 7.3. Shouldn't all columns say 4.17 now?

**Comment by romani** [comment]
4.17 is released only on gihub, to release it sonarqube some other process should pass, see comment - https://github.com/checkstyle/sonar-checkstyle/issues/189#issuecomment-457962480.
Sonar team support this page.

**Comment by Vampire** [comment]
I see, thx

**Pull request merged**